php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java

```diff
@@ -0,0 +1,258 @@
+
+            for (PhpVersion phpVersion : PhpVersion.values()) {
+                model.addElement(phpVersion);
+            }
+
+            phpVersionList.setModel(model);
+            PhpVersion phpVersion = PhpLanguageProperties.forFileObject(fileObject).getPhpVersion();
+            phpVersionList.setSelectedValue(phpVersion, true);
+            phpVersionList.setBorder(new LineBorder(Color.GRAY, 1));
+
+            phpVersionList.addListSelectionListener((ListSelectionEvent e) -> {
+                savePhpVersion(component, phpVersionList.getSelectedValue());
+                showPhpVersion(component.getDocument());
+            });
+            return phpVersionList;
+        }
+
+        void showPhpVersionList() {
+            final JTextComponent component = EditorRegistry.focusedComponent();
+            if (component == null) {
+                Toolkit.getDefaultToolkit().beep();
+                return;
+            }
+            FileObject fileObject = NbEditorUtilities.getFileObject(component.getDocument());
+            if (fileObject == null) {
+                Toolkit.getDefaultToolkit().beep();
+                return;
+            }
+            final JList<PhpVersion> phpVersionList = createPhpVersionList(component, fileObject);
+            final Popup popup = getPopup(phpVersionList);
+            popup.show();
+        }
+
+        private Popup getPopup(JList<PhpVersion> phpVersionList) {
+            Point labelStart = getLocationOnScreen();
+            int x = Math.min(labelStart.x, labelStart.x + getSize().width - phpVersionList.getPreferredSize().width);
+            int y = labelStart.y - phpVersionList.getPreferredSize().height;
+
+            final Popup popup = PopupFactory.getSharedInstance().getPopup(this, phpVersionList, x, y);
+            final AWTEventListener multicastListener = new AWTEventListener() {
+                @Override
+                public void eventDispatched(AWTEvent event) {
+                    if (event instanceof MouseEvent && ((MouseEvent) event).getClickCount() > 0) {
+                        popup.hide();
+                        Toolkit.getDefaultToolkit().removeAWTEventListener(this);
+                    }
+                }
+            };
+            Toolkit.getDefaultToolkit().addAWTEventListener(multicastListener, AWTEvent.MOUSE_EVENT_MASK);
+            return popup;
+        }
+
+        private void savePhpVersion(JTextComponent component, PhpVersion phpVersion) {
+            FileObject fileObject = NbEditorUtilities.getFileObject(component.getDocument());
+            if (fileObject != null) {
+                Project project = FileOwnerQuery.getOwner(fileObject);
+                if (project instanceof PhpProject) {
+                    PhpProjectProperties projectProperties = new PhpProjectProperties((PhpProject) project);
+                    projectProperties.setPhpVersion(phpVersion.name());
+                    projectProperties.save();
```

Comment:
@tmysik I'm not sure whether this is OK/correct for you...

Reply:
It would be perhaps better to save it in a background thread...

Reply:
Exactly.